client/packages/invoices/src/Prescriptions/DetailView/History/HistoryModal.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useMemo } from 'react';
+import {
+  useDialog,
+  useTranslation,
+  ModalMode,
+  TableProvider,
+  createTableStore,
+  createQueryParamsStore,
+  DataTable,
+  InvoiceNodeStatus,
+  useIntlUtils,
+  InvoiceLineNode,
+} from '@openmsupply-client/common';
+import { Draft } from '../../..';
+
+import { useHistoryColumns } from './columns';
+import { usePrescriptionList } from '../../api';
+
+interface HistoryModalModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  draft: Draft | null;
+  mode: ModalMode | null;
+  patientId: string;
+}
+
+export interface HistoryItem {
+  id: string;
+  itemName: string;
+  unitQuantity: number;
+  directions: string;
+  date: Date | null;
+  prescriber: string;
+}
+
+export const HistoryModal: React.FC<HistoryModalModalProps> = ({
+  isOpen,
+  onClose,
+  patientId,
+}) => {
+  const {
+    query: { data, isLoading },
+  } = usePrescriptionList({
+    filterBy: {
+      otherPartyId: { equalTo: patientId },
+      status: { notEqualTo: InvoiceNodeStatus.New },
+    },
+    sortBy: { key: 'pickedDatetime', direction: 'desc' },
+    first: 20,
+  });
+  const columns = useHistoryColumns();
+  const t = useTranslation();
+  const { getLocalisedFullName } = useIntlUtils();
+  const { Modal } = useDialog({ isOpen, onClose, disableBackdrop: true });
+
+  const historyData = useMemo(() => {
+    const rawData = data?.nodes;
+    if (!rawData) return [];
+
+    const compiled: HistoryItem[] = [];
+
+    rawData.forEach(prescription => {
+      const { clinician, pickedDatetime, lines } = prescription;
+      compiled.push(
+        ...combineCommonLines(lines.nodes as InvoiceLineNode[]).map(line => ({
+          id: line.id,
+          itemName: line.itemName,
+          unitQuantity: line.unitQuantity,
+          directions: line.note ?? '',
+          date: pickedDatetime ? new Date(pickedDatetime) : null,
+          prescriber: getLocalisedFullName(
+            clinician?.firstName,
+            clinician?.lastName
+          ),
+        }))
+      );
+    });
+
+    return compiled;
+  }, [data]);
+
+  return (
+    <TableProvider
+      createStore={createTableStore}
+      queryParamsStore={createQueryParamsStore({
+        initialSortBy: { key: 'expiryDate' },
+      })}
+    >
+      <Modal title={t('heading.history')} width={1125} height={600}>
```

Comment:
Assuming this width here might be why it's not going full screen?

Reply:
Yes and no. The `width` and `height` properties are actually converted into `minHeight` and `minWidth` in the Modal component, so they're just minimums really. Have reduced the width somewhat now. 301b8b483a

Have made a separate issue for the lack of responsiveness when re-sizing: https://github.com/msupply-foundation/open-msupply/issues/6311